controller/api/Platforms.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoPublishing\controller\api;
+
+use common_exception_RestApi;
+use oat\taoPublishing\model\entity\Platform;
+use oat\taoPublishing\model\PlatformService;
+use oat\taoPublishing\model\CrudPlatformsService;
+use tao_actions_CommonRestModule;
+use tao_actions_RestController;
+
+/**
+ * @OA\Info(title="Platform API", version="0.1")
+ */
+class Platforms extends tao_actions_CommonRestModule
+{
+    public function __construct()
+    {
+        parent::__construct();
+        $this->service = CrudPlatformsService::singleton();
+    }
+
+    public function index()
+    {
+        $this->returnFailure(new common_exception_RestApi('Not implemented'));
```

Comment:
Maybe it would be better to call parent instead of throwing an exception.

Reply:
With the current implementation of `oat\taoPublishing\model\routing\ApiRoute` it's not possible to make an API request that would invoke `index` method. But in case it would be changed, I am making sure that parent `put`/`post`/`delete` will not be called, since I did not properly implement/test them.

Reply:
probably would be better to change it to:

```
public function index() {
  $request = $this->getPsrRequest();
  if ($request->getMethod() !== 'GET') {
    $this->returnFailure(new common_exception_BadRequest($request->getUri()->getPath()));
  }

  parent::index();
}
```

I guess it is less complex (but it is for me, so am waiting for your opinion :) )